index.d.ts

```diff
@@ -0,0 +1,249 @@
+     */
+    readonly isRunning: boolean;
+
+    /**
+     * `true` if any future task instances are queued.
+     */
+    readonly isQueued: boolean;
+
+    /**
+     * `true` if the task is not in the running or queued state.
+     */
+    readonly isIdle: boolean;
+
+    /**
+     * The current state of the task: `"running"`, `"queued"` or `"idle"`.
+     */
+    readonly state: TaskState;
+
+    /**
+     * The most recently started task instance.
+     */
+    readonly last: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recent task instance that is currently running.
+     */
+    readonly lastRunning: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recently performed task instance.
+     */
+    readonly lastPerformed: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recent task instance that succeeded.
+     */
+    readonly lastSuccessful: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recently completed task instance.
+     */
+    readonly lastComplete: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recent task instance that errored.
+     */
+    readonly lastErrored: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recently canceled task instance.
+     */
+    readonly lastCanceled: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The most recent task instance that is incomplete.
+     */
+    readonly lastIncompleted: TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * The number of times this task has been performed.
+     */
+    readonly performCount: number;
+
+    readonly numRunning: number;
+    readonly numQueued: number;
+
+    /**
+     * alias for `numRunning`
+     */
+    readonly concurrency: number;
+
+    readonly context: any;
+
+    readonly name: string;
+
+    perform(...args: PerformArgs): TaskInstance<PerformArgs, PerformReturn>;
+
+    /**
+     * Cancels all running or queued `TaskInstance`s for this Task.
+     * If you're trying to cancel a specific TaskInstance (rather
+     * than all of the instances running under this task) call
+     * `.cancel()` on the specific TaskInstance.
+     */
+    cancelAll(): void;
+  }
+}
+
+declare module 'ember-concurrency/task-instance' {
+  import Task from 'ember-concurrency/task';
+
+  export enum TaskInstanceState {
+    /** task instance was canceled before it started */
+    DROPPED = 'dopped',
+
+    /** task instance was canceled before it could finish */
+    CANCELED = 'canceled',
+
+    /** task instance ran to completion (even if an exception was thrown) */
+    FINISHED = 'finished',
+
+    /** task instance is currently running (returns true even if is paused on a yielded promise) */
+    RUNNING = 'running',
+
+    /** task instance hasn't begun running yet (usually because the task is using the `TaskProperty#enqueue` task modifier) */
+    WAITING = 'waiting',
+  }
```

Comment:
same deal here, re: exporting of an enum

Reply:
yeah, these probably don't need to be exported -- this is _mostly_ copy paste from long ago